webapp/publisher/snaps/views.py

```diff
@@ -235,6 +235,16 @@ def get_listing_snap(snap_name):
         m["url"] for m in snap_details["media"] if m["type"] == "screenshot"
     ]
 
+    licenses = []
+    for license in get_licenses():
+        licenses.append({"key": license["licenseId"], "name": license["name"]})
+
+    license = snap_details["license"]
+    license_type = "custom"
+
+    if "AND" not in license.upper() and "WITH" not in license.upper():
```

Comment:
Not sure if this may be the case right now, but if any license name/code contains 'AND' or 'WITH' this check would fail.

Reply:
hmm, there are some licenses that include "WITH", good catch. a bit hackey but changing this to `if " AND " ... and " WITH "...` should fix this potential issue